### PR TITLE
Change performance miniapps compiler flags for Apple M1

### PR DESCRIPTION
**Pull request by pazner**
Related to #2273.

Should help resolve mfem/pymfem#113.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge|
 | --- | --- | --- | --- | --- | --- | --- |
| [#3233](https://github.com/mfem/mfem/pull/3233) | @pazner | @tzanio | @camierjs + @v-dobrev | 9/27/22 | 10/7/22 | 10/14/22 | |

<details>
<summary>PR Checklist</summary>
    
- [ ] Code builds.
- [ ] Code passes `make style`.
- [ ] Update `CHANGELOG`:
    - [ ] Is this a new feature users need to be aware of? New or updated example or miniapp?
    - [ ] Does it make sense to create a new section in the `CHANGELOG` to group with other related features?
- [ ] Update `INSTALL`:
    - [ ] Had a new optional library been added? If so, what range of versions of this library are required? (*Make sure the external library is compatible with our BSD license, e.g. it is not licensed under GPL!*)
    - [ ] Have the version ranges for any required or optional libraries changed?
    - [ ] Does `make` or `cmake` have a new target?
    - [ ] Did the requirements or the installation process change? *(rare)*
- [ ] Update continuous integration server configurations if necessary (e.g. with new version requirements for each of MFEM's dependencies)
    - [ ] `.github`
    - [ ] `.appveyor.yml`
- [ ] Update `.gitignore`:
    - [ ] Check if `make distclean; git status` shows any files that were generated from the source by the project (not an IDE) but we don't want to track in the repository.
    - [ ] Add new patterns (just for the new files above) and re-run the above test.
- [ ] New examples:
    - [ ] All sample runs at the top of the example source file work.
    - [ ] Update `examples/makefile`:
      - [ ] Add the example code to the appropriate `SEQ_EXAMPLES` and `PAR_EXAMPLES` variables.
      - [ ] Add any files generated by it to the `clean` target.
      - [ ] Add the example binary and any files generated by it to the top-level `.gitignore` file.
    - [ ] Update `examples/CMakeLists.txt`:
      - [ ] Add the example code to the `ALL_EXE_SRCS` variable.
      - [ ] Make sure `THIS_TEST_OPTIONS` is set correctly for the new example.
   - [ ] List the new example in `doc/CodeDocumentation.dox`.
   - [ ] If new examples directory (e.g.`examples/pumi`), list it in `doc/CodeDocumentation.conf.in`
   - [ ] Companion pull request for documentation in [mfem/web](https://github.com/mfem/web) repo:
      - [ ] Update or add example-specific documentation, see e.g. the `src/examples.md`.
      - [ ] Add the description, labels and screenshots in `src/examples.md` and `src/img`.
      - [ ] In `examples.md`, list the example under the appropriate categories, add new categories if necessary.
      - [ ] Add a short description of the example in the "Extensive Examples" section of `features.md`.
- [ ] New miniapps:
   - [ ] All sample runs at the top of the miniapp source file work.
   - [ ] Update top-level `makefile` and `makefile` in corresponding miniapp directory.
   - [ ] Add the miniapp binary and any files generated by it to the top-level `.gitignore` file.
   - [ ] Update CMake build system:
      - [ ] Update the `CMakeLists.txt` file in the `miniapps` directory, if the new miniapp is in a new directory.
      - [ ] Add/update the `CMakeLists.txt` file in the new miniapp directory.
      - [ ] Consider adding a new test for the new miniapp.
   - [ ] List the new miniapp in `doc/CodeDocumentation.dox`
   - [ ] If new miniapps directory (e.g.`miniapps/nurbs`), add it to `MINIAPP_SUBDIRS` in the `makefile`.
   - [ ] If new miniapps directory (e.g.`miniapps/nurbs`), list it in `doc/CodeDocumentation.conf.in`
   - [ ] Companion pull request for documentation in [mfem/web](https://github.com/mfem/web) repo:
     - [ ] Update or add miniapp-specific documentation, see e.g. the `src/meshing.md` and `src/electromagnetics.md` files.
     - [ ] Add the description, labels and screenshots in `src/examples.md` and `src/img`.
     - [ ] The miniapps go at the end of the page, and are usually listed only under a specific "Application (PDE)" category.
     - [ ] Add a short description of the miniapp in the "Extensive Examples" section of `features.md`.
- [ ] New capability:
   - [ ] All new public, protected, and private classes, methods, data members, and functions have full Doxygen-style documentation in source comments. Documentation should include descriptions of member data, function arguments and return values, template parameters, and prerequisites for calling new functions.
   - [ ] Pointer arguments and return values must specify whether ownership is being transferred or lent with the call.
   - [ ] Any new functions should include descriptions of their intended use e.g. for internal use only, user-facing, etc., along with references to example code whenever possible/appropriate.
   - [ ] Consider adding new sample runs in existing examples to highlight the new capability.
   - [ ] Consider saving cool simulation pictures with the new capability in the Confluence gallery (LLNL only) or submitting them, via pull request, to the gallery section of the `mfem/web` repo.
   - [ ] If this is a major new feature, consider mentioning it in the short summary inside `README` *(rare)*.
   - [ ] List major new classes in `doc/CodeDocumentation.dox` *(rare)*.
- [ ] Update this checklist, if the new pull request affects it.
- [ ] Run `make unittest` to make sure all unit tests pass.
- [ ] Run the tests in `tests/scripts`.
- [ ] (LLNL only) After merging:
   - [ ] Update internal tests to include the new features.
</details>

### CODE REVIEW

**Review by tzanio — APPROVED** [review]

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Comment by tzanio** [comment]
Can someone else with access to an M1 mac try this?

**Comment by tbridel** [comment]
Hello everyone at MFEM 👋 
/cc @tzanio @pazner 

I just tried this fix and it works on my platform (macOS Monterey 12.4 with Apple M1 Pro).

Before the fix:

```bash
...
[ 84%] Building CXX object miniapps/performance/CMakeFiles/performance_ex1.dir/ex1.cpp.o
[ 85%] Linking CXX executable shaper
clang: error: the clang compiler does not support '-march=native'
make[2]: *** [miniapps/performance/CMakeFiles/performance_ex1.dir/ex1.cpp.o] Error 1
make[1]: *** [miniapps/performance/CMakeFiles/performance_ex1.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
...
```

After the fix:

```bash
...
[100%] Linking CXX executable performance_ex1
[100%] Built target performance_ex1
[100%] Built target miniapps
...
```

Thanks guys !

**Comment by pazner** [comment]
Thanks for checking, @tbridel!

**Comment by tbridel** [comment]
> Thanks for checking, @tbridel!

By the way @pazner (/cc @tzanio), do you think there is any way to go ahead with the merge into `master` earlier than planned ? 
I am working with PyMFEM and I would love the `master` of MFEM to include this fix so that `pip install mfem` works seamlessly on Darwin platforms 😍

**Comment by pazner** [comment]
Hi @tbridel, we will try to merge this as quickly as possible. We should just get approvals from @v-dobrev and @camierjs first. I agree, getting `pip install mfem` to work on M1 Macs is a priority 😄

**Review by camierjs — APPROVED** [review]

File: miniapps/performance/CMakeLists.txt [hunk 1]
```diff
@@ -11,21 +11,22 @@
...
   list(APPEND PERFORMANCE_CXX_OPTIONS
-    "-march=native"
+    ${MFEM_PERF_CXX_ARCH_FLAGS}
     "-fcolor-diagnostics"
     "-fvectorize"
     "-fslp-vectorize"
     "-ffp-contract=fast")
 elseif (${CMAKE_CXX_COMPILER_ID} MATCHES "GNU")
   list(APPEND PERFORMANCE_CXX_OPTIONS
-    ${MFEM_PERF_CXX_ARCH_FLAGS}
```
Comment (v-dobrev) [inline miniapps/performance/CMakeLists.txt:28]: Why remove this line? This handles the GCC compiler.

Comment (pazner) [reply miniapps/performance/CMakeLists.txt]: I think you're right… don't know why I made that change. I reverted it.

**Review by v-dobrev — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Comment by CsatiZoltan** [comment]
I can testify that with the [modification](https://github.com/mfem/mfem/pull/3233/files), I do not get building errors.
OS: macOS Monterey
Chip: Apple M1
Kernel: Darwin Kernel version 21.6.0

**Pull request merged**